experimentation/src/main/java/com/automattic/android/experimentation/ExPlat.kt

```diff
@@ -74,20 +81,28 @@ class ExPlat(
     }
 
     private fun getAssignments(refreshStrategy: RefreshStrategy): Assignments {
```

Comment:
Question (❓): Please help me understand a bit more about this implementation here and how it actually being/should be utilized by any client of this library. 🙏 

1. To my understanding this function returns the (previously) cached assignments no matter what, then why is it fetching the assignment if refresh strategy is the right one or the assignment are stale? I really struggle to understand the logic here. I know you didn't write this, and only want to change that `FluxC` implementation 1:1 to what a `non-FluxC` should look like, but still, how does this `ExPlat` work in general. What I was expecting during that call is for the fetched assignments to be returned instead of the cached ones when the strategy/stale conditions are met. 🤔 
2. I then tried to research that myself real quick, searching the `ExPlat` keyword on both [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android) and [WCAndroid](https://github.com/woocommerce/woocommerce-android) and figured out that:
    1. [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android): This client is actually using its own version of `ExPlat`, right! 🤷 But, it shouldn't, should it? 😕 Maybe we need to update this client when we are ready with the `Tracks` new implementation, wdyt?  🤔
    2. [WCAndroid](https://github.com/woocommerce/woocommerce-android): This client is using the `Tracks` version of `ExPlat` but actually not utilizing it at all, right! 🤷 Is it actually needed there?
    3. To my understanding, no other client is using the `Tracks` version of `ExPlat`, just because of `FluxC`.

Reply:
Good questions! I had similar confusions and pivots in understanding this.

Trying to answer broadly: we want `ExPlat` to not return different values for the same application session. So we initially return cached value and, under some circumstances, we request an update of the cache.

The update will be used only on the next application session or after app calls `ExPlat#clear` method (if e.g. user logs out).

**Why we don't want to return different values for the same application session?**

We want to protect SDK clients from a mistake, in which the would e.g. run half of the app processes in a `Control` group, then have an update of cache, and the rest of the app in `Treatment` group. This would lead to unexpected behaviors and failed experiments.

**Why we don't just fetch newest experiment assignment?**

We want this SDK to be async and not block clients with a `suspend` calls as some experiment changes might require e.g. different app initialization.

---

I hope this clarifies things. For the complete summary and explanations of the final design, please see #243 - I think everything might be more readable after the lecture.

---

> I then tried to research that myself real quick, searching the ExPlat keyword on both [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android) and [WCAndroid](https://github.com/woocommerce/woocommerce-android)

Yes, to my best knowledge, no Android mobile project at Automattic is doing A/B tests using ExPlat at this moment

Reply:
Awesome, thank you so much for clarifying my question(s) @wzieba , much much appriciated. Also, thanks so much for #243 and the `README.md` you are adding there, haven't got that far with my reviews yet, but I am sure this will further clarify any doubts I have about this library, it's public APIs, including how to use it.

> Yes, to my best knowledge, no Android mobile project at Automattic is doing A/B tests using ExPlat at this moment

This is a bit weird to me, if not risky... I wish we knew that this library is working as expected and that we have thoroughly tested it with some clients before having other clients using it, assuming it just works... 🤔

I know, maybe that library worked at some point on [WCAndroid](https://github.com/woocommerce/woocommerce-android), they are just not using it anymore, not having an experiment worth of this utility, but still, as time passes-by, having a stale unmaintained/untested library might create some weird issues we might not/cannot be aware of atm.

You addition a new set of buttons/fields (an experimentation screen) as part of #241 mitigates this risk for us, as we at least can verify that we see it working, which gives us confidence, but I want to raise that thought of mine anyway, that we should be careful and warn product teams about this library and it being "untested" to its fullest. Am I being paranoid here? 😅

Reply:
Not at all! I think we should request a safety check from product teams on this and, for now, release this as a `"beta"`.

Reply:
Cool, then totally agree with your plan! 🎯